### PR TITLE
CR-1135275 Xbmgmt should not auto-retry SC update, on failure

### PR DESCRIPTION
**Pull request by rchane**
#### Problem solved by the commit
https://jira.xilinx.com/browse/CR-1135275 
In dmesg, one can see that xbmgmt tries to update SC again after an initial failure.
#### Bug / issue (if any) fixed, which PR introduced the bug, how it was discovered
There was an issue with flushing. 
#### How problem was solved, alternative solutions (if any) and why they were rejected
`fs.flush()` was changed to `fs.close()` to prevent the stream from trying to write again to SC after `put` returned.
#### Risks (if any) associated the changes in the commit
N/A
#### What has been tested and how, request additional testing if necessary
Testing was done on a vck5000 with a xsabin that forced SC update failure (can be found in the CR). As this is a priority 1 CR, I request that the verifier also tested this themselves for normal and failed SC updates, please.

dmesg using flush():
```
[...]
[82593.059210] audit: type=1400 audit(1662672301.327:562): apparmor="ALLOWED" operation="open" profile="/usr/sbin/sssd" name="/proc/107677/cmdline" pid=1270 comm="sssd_nss" requested_mask="r" denied_mask="r" fsuid=0 ouid=0
[82624.493222] xclmgmt 0000:65:00.0: xgq_vmr.m.54525952 ffff9e6486620010 xgq_vmr_log_dump_debug: opcode: PROGRAM SCFW(0xf), rcode: 1, check debug trace for detailed log.
[82624.493712] xclmgmt 0000:65:00.0: xgq_vmr.m.54525952 ffff9e6486620010 program_sc_store: failed: 1
[82652.533337] xclmgmt 0000:65:00.0: xgq_vmr.m.54525952 ffff9e6486620010 xgq_vmr_log_dump_debug: opcode: PROGRAM SCFW(0xf), rcode: 1, check debug trace for detailed log.
[82652.533817] xclmgmt 0000:65:00.0: xgq_vmr.m.54525952 ffff9e6486620010 program_sc_store: failed: 1
```
dmesg using close():
```
[...]
[82974.158725] xocl 0000:17:00.1:  ffff9e6487f790b0 xocl_drvinst_close: CLOSE 2
[82974.158730] xocl 0000:17:00.1:  ffff9e6487f790b0 xocl_drvinst_close: NOTIFY 00000000cb10d1b6
[83097.575245] xclmgmt 0000:65:00.0: xgq_vmr.m.54525952 ffff9e6486620010 xgq_vmr_log_dump_debug: opcode: PROGRAM SCFW(0xf), rcode: 1, check debug trace for detailed log.
[83097.575709] xclmgmt 0000:65:00.0: xgq_vmr.m.54525952 ffff9e6486620010 program_sc_store: failed: 1
```
#### Documentation impact (if any)
N/A

### CODE REVIEW

File: src/runtime_src/core/pcie/linux/scan.cpp [hunk 1]
```diff
@@ -318,7 +318,7 @@ put(const std::string& name,
   if (!err.empty())
     return;
   fs << input;
-  fs.flush();
+  fs.close();
   if (!fs.good()) {
```
Comment (stsoe) [inline src/runtime_src/core/pcie/linux/scan.cpp:322]: It doesn't make any sense to me to check fs.good() after closing the stream.
Please clarify.

Comment (rchane) [reply src/runtime_src/core/pcie/linux/scan.cpp:322]: My mistake; you are right. I didn't see that fs.good() became 0 after closing the stream.

**Comment by dayeh-xilinx** [comment]
retest this please.

File: src/runtime_src/core/pcie/linux/scan.cpp (hunk 1 above)
Comment (stsoe) [reply src/runtime_src/core/pcie/linux/scan.cpp:322]: Actually Ryan, checking good() after close() is correct.  Just looked strange at first glance, so maybe a comment would be appropriate, something like:
`fs.close();  // flush and close, if either fails then stream failbit is set.`

Comment (rchane) [reply src/runtime_src/core/pcie/linux/scan.cpp]: Did so, thank you!

**Review by stsoe — APPROVED** [review]

**Pull request merged**